catroid/src/main/java/org/catrobat/catroid/transfers/ProjectUploadService.java

```diff
@@ -122,7 +125,14 @@ protected void onHandleIntent(Intent intent) {
 			}
 
 			File archive = new File(CACHE_DIR, UPLOAD_FILE_NAME);
-			new ZipArchiver().zip(archive, projectDir.listFiles());
+			ArrayList<File> files = new ArrayList<>(Arrays.asList(projectDir.listFiles()));
+			for (File f: files) {
```

Comment:
you could just create a File deviceVarFile = new File(DEVICE_VARIABL....)
and use files.remove(deviceVarFile).

file equals works on abstract path:
     * Tests this abstract pathname for equality with the given object.
     * Returns <code>true</code> if and only if the argument is not
     * <code>null</code> and is an abstract pathname that denotes the same file
     * or directory as this abstract pathname.  Whether or not two abstract
     * pathnames are equal depends upon the underlying system.  On UNIX
     * systems, alphabetic case is significant in comparing pathnames; on Microsoft Windows
     * systems it is not.